mdata/cache/ccache_metric.go

```diff
@@ -69,6 +69,66 @@ func (mc *CCacheMetric) Del(ts uint32) int {
 	return len(mc.chunks)
 }
 
+func (mc *CCacheMetric) AddSlice(prev uint32, itergens []chunk.IterGen) {
```

Comment:
this function should document its assumptions/requirements, which afaik is two-fold (correct me if i'm wrong)
* the sequence of chunks should be in ascending timestamp order
* the sequence should be complete (no gaps)

also I don't like the name AddSlice, slice is just an implementation detail, it doesn't describe the concept well enough.
Perhaps AddSequence, AddWindow or AddRange is better.